### PR TITLE
Use +- install output for Python versions

### PR DESCRIPTION
**Pull request by charliermarsh**
## Summary

Follow-up to https://github.com/astral-sh/uv/pull/4939. Uses a format that's closer to `uv pip install`, with some special-casing for single Pythons.

## Test Plan

A few examples:

Screenshot 2024-07-18 at 8 39 35 PM
Screenshot 2024-07-18 at 8 39 46 PM
Screenshot 2024-07-18 at 8 39 27 PM
Screenshot 2024-07-18 at 8 39 54 PM

### CODE REVIEW

File: crates/uv/src/commands/python/uninstall.rs [hunk 1]
```diff
@@ -68,18 +69,7 @@ pub(crate) async fn uninstall(
...
-                    writeln!(printer.stderr(), "Found: {}", installation.key().green(),)?;
-                } else {
-                    writeln!(
-                        printer.stderr(),
-                        "Found existing installation for {}: {}",
-                        request.cyan(),
-                        installation.key().green(),
-                    )?;
-                }
-            }
```
Comment (charliermarsh) [inline crates/uv/src/commands/python/uninstall.rs:82]: This ended up feeling unnecessary now that we show the summary at the end.

File: crates/uv-python/src/installation.rs [hunk 2]
```diff
@@ -350,7 +350,7 @@ impl Ord for PythonInstallationKey {
     fn cmp(&self, other: &Self) -> std::cmp::Ordering {
         self.implementation
             .cmp(&other.implementation)
-            .then_with(|| other.version().cmp(&self.version()))
+            .then_with(|| self.version().cmp(&other.version()))
```
Comment (charliermarsh) [inline crates/uv-python/src/installation.rs:353]: I think this is inverted.

**Review by zanieb — APPROVED** [review]
Gorgeous.

**Comment by charliermarsh** [comment]
Thank you!

File: crates/uv-python/src/installation.rs (hunk 2 above)
Comment (j178) [reply crates/uv-python/src/installation.rs:353]: Sorry I should've added a explanation here. This is intentional to ensure that Python installations are sorted in descending order by version. https://github.com/astral-sh/uv/issues/5139

Comment (zanieb) [reply crates/uv-python/src/installation.rs:353]: Uh oh! haha

Comment (charliermarsh) [reply crates/uv-python/src/installation.rs:353]: Hah, I'll take a look.

**Pull request merged**